wolfcrypt/src/integer.c

```diff
@@ -1751,6 +1755,10 @@ int s_mp_add (mp_int * a, mp_int * b, mp_int * c)
       *tmpc++ &= MP_MASK;
     }
 
+    /* ensure pointer still points to valid memory before proceeding */
+    if (!tmpc)
```

Comment:
Obviously this should not be necessary -- the nullness check on `c->dp` precludes `tmpc` being null.

If an analyzer is complaining here, that sounds like a false positive.

Reply:
If the pointer always pointed to the same address I would agree.
Line 1755 changes the address the pointer is pointing to.
I do not think it is a "bad thing" to ensure the new address is still a valid address after the address has been modified multiple times in the for loop.

Reply:
Alternative: To consider this a false positive then we must consider both clang-10 and clang-12 to be broken (which both report on this item) and remove the test for this build that used to pass in favor of just not testing it. Let me know.